src/main/java/com/minecraftabnormals/upgrade_aquatic/common/entities/jellyfish/helper/JellyfishSizeMap.java

```diff
@@ -0,0 +1,117 @@
+package com.minecraftabnormals.upgrade_aquatic.common.entities.jellyfish.helper;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class JellyfishSizeMap {
+    protected final TreeMap<Float, Integer> sizeMap;
+    protected final int weightTotal;
+
+    public JellyfishSizeMap(TreeMap<Float, Integer> map) {
+        this.sizeMap = map;
+        int weightTotal = 0;
+        for (int weight : this.sizeMap.values())
+            weightTotal += weight;
+        this.weightTotal = weightTotal;
+    }
+
+    public int size() {
+        return sizeMap.size();
+    }
+
+    public int totalWeight() {
+        return this.weightTotal;
+    }
+
+    public Float randomSize(Random rand) throws RuntimeException {
+        int denominator = totalWeight();
+        for (Map.Entry<Float, Integer> sizeEntry : this.entrySet()) {
+            int weight = sizeEntry.getValue();
+            if (rand.nextInt(denominator) < weight) {
+                return sizeEntry.getKey();
+            }
+            denominator = denominator - weight;
+        }
+       throw new RuntimeException("JellyfishSizeMap in Upgrade Aquatic has broken. This should not be happening!");
```

Comment:
I don't like this exception being thrown here.
Just have it return `1.0F` instead.

Reply:
I guess, but if the return value is never reached it either means the total weight has been fucked with or the inputted `Random` is weird. I don't know if that should just be ignored.

Reply:
I don't think an issue caused by a `Random` being faulty should be accounted for because that's an issue caused by the person manipulating the `Random`.
As for the total weight. If there are certain conditions the total weight must meet, it should get checked during the construction of `JellyfishSizeMap`

Reply:
That's fair. The total weight is assigned in the constructor so it could only be messed up if you input an empty list, so I guess that could be checked in the constructor. I'll have to leave it to the morning though, as it's getting a bit late. :P

Reply:
I made the constructor throw a non-fatal exception if you input an empty list. The problem with this is that it appears in the log when the static variable is initialised, and that's usually far before you'll try to spawn a jellyfish, so it would be easy to miss. Of course, it's not like this is a public library so it probably doesn't matter.